### ISSUE TITLE
Document templates ignored in flow with "send mail"

### ISSUE THREAD

**Issue by DanielPa** [dneustadt/DneTemplateManager#8]
I created a custom template for `/base.html.twig` that works great as far as I create, lets say a invoice through an orders details in the backend. 
Also if I create a flow that has only the `Generate document` action in the action block, the resulting PDF looks like expected with the customization from the template. But when I now add a `Send email` action to the block, right behind the `Generate document`, the generated document seems to ignore all custom templates. 
Could this be an extension issue?

BTW: Thanks for this great plugin 🤩

**Comment by dneustadt** [comment]
Hmm, yeah that's weird. The send mail action only takes documents already generated when it's executed and should not influence the actions that were executed prior to that. Is it just the attached documents that ignore the extension or does the send mail action somehow also affect the generation in the file system? Also please let me know which Shopware version you're on and I'll look into it.

**Comment by DanielPa** [comment]
Thanks for your quick response. 
It affects both documents, the sent one and the one on the filesystem (which should be the same)
Shopware Version is `6.4.13.0`

I also tried it with two separate flows:
1. Flow: trigged on a `Checkout / Order / Placed` if `MyRule == true`=> `Generate document`+ `Set states` (`Paid`/`Shipped`/`Completed`)
2. Flow: trigged on `State Enter / Order / State / Completed` => `Send email`

Result was the same like with my first try all in a single flow (custom template had no effect)
If I disable only the send email flow, the document is generated correctly.

**Comment by DanielPa** [comment]
I tried to follow the code... well I'm anything else but an php expert but I see a problem in the  SendMailAction that uses its own DocumentService an then calls getDocument [here](https://github.com/shopware/platform/blob/ca7537bb61cf0306be50ce312ed2b5cff66e151b/src/Core/Content/Flow/Dispatching/Action/SendMailAction.php#L437). 
And the getDocument (in a condition I can't follow) calls generateDocument [here](https://github.com/shopware/platform/blob/ca7537bb61cf0306be50ce312ed2b5cff66e151b/src/Core/Checkout/Document/DocumentService.php#L192) which seems to result in a new file. 

But maybe I missed something and I'm completely wrong 🤷‍♂️

**Comment by dneustadt** [comment]
Your findings are correct. It seems to be a bug in the core. The action for generating documents uses a different method to create documents that respects template inheritance. This seems to have been fixed for `6.5.0.0` already, as there is a change behind a feature flag in `SendMailAction`:

```php
if (Feature::isActive('v6.5.0.0')) {
    // this fetches existing documents
    $attachments = $this->mappingAttachments($documentIds, $attachments, $mailEvent->getContext());
} else {
    // this generates documents (without inherited template changes)
    $attachments = $this->buildOrderAttachments($documentIds, $attachments, $mailEvent->getContext());
}
```

I think the reason is that beginning with `6.5.0.0` a document generation action is required before a send mail action, while currently it is not. I'm not sure if this can be fixed right now as they will probably tell you to wait for the next major release. Until that release you could replace the if/else construct with just the line `$attachments = $this->mappingAttachments($documentIds, $attachments, $mailEvent->getContext());` and see if this works (looks to me like it should). Sorry I can't be of great help here but since all these methods already are deprecated, I don't think trying to fix them now is viable when the next major is not that far away.

**Comment by DanielPa** [comment]
Thanks for helping, although it had nothing to do with your plugin at all!
